### ISSUE TITLE
Metrics results ToString overload?

### ISSUE THREAD

**Issue by TomFinley** [dotnet/machinelearning#2053]
Consider this class, a result from a non-calibrated evaluation of binary classification.

https://github.com/dotnet/machinelearning/blob/e2f7075782764c9dab8f0a04013aaf8b9921d984/src/Microsoft.ML.Data/Evaluators/Metrics/BinaryClassificationMetrics.cs#L10

It might be nice if `ToString` was enabled on it, would it not? Maybe something that looks like a JSON serialized object (perhaps even exploiting the presence of Newtonsoft, since we must rely on it elsewhere?) would be appropriate? Who knows.

Needless to say, if we do it for one we should probably do it for all.

**Comment by PaulTFreedman** [comment]
@justinormont What data type are you thinking the ToJson() method would return? Something built in to .NET or maybe from Newtonsoft?

**Comment by justinormont** [comment]
Since the class has basic doubles, I would expect nothing fancy is needed for serialization.

The .ToJSON() for [BinaryClassificationMetrics](https://github.com/dotnet/machinelearning/blob/e2f7075782764c9dab8f0a04013aaf8b9921d984/src/Microsoft.ML.Data/Evaluators/Metrics/BinaryClassificationMetrics.cs#L10) could look like:
```json
{
  "Auc": 0.95, 
  "Accuracy": 0.94, 
  "PositivePrecision": 0.93, 
  "PositiveRecall": 0.92, 
  "NegativePrecision": 0.91, 
  "NegativeRecall": 0.9, 
  "F1Score": 0.89, 
  "Auprc": 0.88 
}
```

**Comment by PaulTFreedman** [comment]
Let me rephrase the question. I have added a `ToString` override like so to `BinaryClassificationMetrics`:
```
